### PR TITLE
CLN: test and repo structure

### PR DESCRIPTION
**Pull request by chiahaoliu**
Clean repo structure, add rever and fix py3 syntax issues.

### CODE REVIEW

**Comment by chiahaoliu** [comment]
@CJ-Wright @cfarrow @pavoljuhas 

Feel free to review if you have time.

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ before_install:
   - chmod +x miniconda.sh
   - ./miniconda.sh -b -p ~/mc
   - export PATH=~/mc/bin:$PATH
+  - conda config --set always_yes yes --set changeps1 no --set quiet true
   - conda update conda --yes
   - conda config --add channels conda-forge
```
Comment (CJ-Wright) [inline .travis.yml]: Can you move this up before conda update conda?

File: .travis.yml [hunk 2]
```diff
@@ -35,9 +37,12 @@ script:
...
   - coverage report -m
   - codecov
-  - flake8 diffpy
   - conda install --yes --file requirements/docs.txt
   - cd docs
   - make html
   - cd ..
   - doctr deploy . --built-docs ./docs/_build/html;
+
+after-script:
```
Comment (CJ-Wright) [inline .travis.yml]: after_script

File: diffpy/pdfmorph/pdfmorphapp.py [hunk 3]
```diff
@@ -245,8 +245,8 @@ def main():
         if opts.savefile == "-":
             outfile = sys.stdout
         else:
-            outfile = file(opts.savefile, 'w')
-        print >> outfile, header
+            outfile = open(opts.savefile, 'w')
+        print(header, file=outfile)
         import numpy
```
Comment (CJ-Wright) [inline diffpy/pdfmorph/pdfmorphapp.py]: Should this be at the top?

File: rever.xsh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+$VERSION_BUMP_PATTERNS = [
+    ('diffpy/pdfmorph/__init__.py', '__version__\s*=.*', "__version__ = '$VERSION'"),
+    ('setup.py', 'version\s*=.*,', "version='$VERSION',")
+    ]
+$CHANGELOG_FILENAME = 'CHANGELOG.rst'
+$CHANGELOG_TEMPLATE = 'TEMPLATE.rst'
+$PUSH_TAG_REMOTE = 'git@github.com:diffpy/diffpy.pdfmorph.git'
+
+$GITHUB_ORG = 'regro'
+$GITHUB_REPO = 'rever'
```
Comment (CJ-Wright) [inline rever.xsh]: We might want to add a licence statement like in the XPD stack, eg https://github.com/xpdAcq/xpdtools/releases/tag/0.1.8

@sbillinge

File: diffpy/pdfmorph/pdfmorphapp.py (hunk 3 above)
Comment (chiahaoliu) [reply diffpy/pdfmorph/pdfmorphapp.py]: @pavoljuhas any insight?

File: .travis.yml [hunk 5]
```diff
@@ -10,17 +10,17 @@ python:
...
   - 3.6
+    env: BULD_DOCS=1
+
 before_install:
   - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh
   - chmod +x miniconda.sh
   - ./miniconda.sh -b -p ~/mc
   - export PATH=~/mc/bin:$PATH
-  - conda config --add channels conda-forge diffpy/channel/dev
+  - conda config --add channels diffpy/channel/dev conda-forge
```
Comment (CJ-Wright) [inline .travis.yml]: Why did the diffpy channel get added?

File: diffpy/pdfmorph/pdfmorphapp.py (hunk 3 above)
Comment (pavoljuhas) [reply diffpy/pdfmorph/pdfmorphapp.py]: Yes, I would move `import numpy` to the  top.

@chiahaoliu - please apply this patch as numpy.savetxt does not work with zip iterator
```diff
diff --git a/diffpy/pdfmorph/pdfmorphapp.py b/diffpy/pdfmorph/pdfmorphapp.py
index 851ddd9..a692453 100755
--- a/diffpy/pdfmorph/pdfmorphapp.py
+++ b/diffpy/pdfmorph/pdfmorphapp.py
@@ -248,7 +248,8 @@ def main():
             outfile = open(opts.savefile, 'w')
         print(header, file=outfile)
         import numpy
-        numpy.savetxt(outfile, zip(chain.xobjout, chain.yobjout))
+        numpy.savetxt(outfile,
+                      numpy.transpose([chain.xobjout, chain.yobjout]))
         outfile.close()
 
     if opts.plot:
```

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+cache:
+  directories:
+    - $HOME/.cache/pip
+    - $HOME/.cache/matplotlib
+
+python:
+  - 2.7
+  - 3.5
+  - 3.6
+    env: BUILD_DOCS=1
```
Comment (CJ-Wright) [inline .travis.yml]: I'd remove this (it may be the source of the travis issue) and just parse the `$TRAVIS_PYTHON_VERSION` env var.

**Comment by chiahaoliu** [comment]
@pavoljuhas would it make sense to have another release of ``diffpy.utils`` for python 3 version? It seems python 3 build on is missing on ``conda-forge`` but I was aware there is a ``v3.0a2`` release from the repo. Thanks.

**Comment by pavoljuhas** [comment]
> would it make sense to have another release of diffpy.utils for python 3 version?

Yes, I suppose so.  There is a very small difference between the py2 and py3 branches of diffpy.utils - I think they can be merged together with a bit of help from the `six` package.  

Can you create an issue for that for diffpy.utils?

**Comment by chiahaoliu** [comment]
Sure!

**Comment by CJ-Wright** [comment]
@pavoljuhas pinging this

**Comment by CJ-Wright** [comment]
This seems to be passing tests and such. I'm going to merge this to keep things moving.

**Pull request merged**